erts/emulator/nifs/common/prim_file_nif.c

```diff
@@ -1399,3 +1404,80 @@ static ERL_NIF_TERM altname_nif(ErlNifEnv *env, int argc, const ERL_NIF_TERM arg
 
     return enif_make_tuple2(env, am_ok, result);
 }
+
+static ERL_NIF_TERM copy_file_range_available_nif(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[]) {
+    off64_t ret;
+
+    ret = syscall(SYS_copy_file_range, -1, NULL, -1, NULL, 0, 0);
+
+    if (ret == -1 && errno == ENOSYS) {
+        return enif_make_atom(env, "false");
+    } else {
+        return enif_make_atom(env, "true");
+    }
+}
+
+static ERL_NIF_TERM copy_file_range_nif(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[]) {
+    posix_errno_t posix_errno, ignored;
+    efile_data_t *d_in, *d_out;
+    off64_t ret = 0, len;
+    efile_fileinfo_t info = {0};
+    efile_path_t file_path_in, file_path_out;
+    char atom_name[256];
+
+    ASSERT(argc == 3);
+
+    if ((posix_errno = efile_marshal_path(env, argv[0], &file_path_in))) {
+        return posix_error_to_tuple(env, posix_errno);
+    }
+
+    if ((posix_errno = efile_marshal_path(env, argv[1], &file_path_out))) {
+        return posix_error_to_tuple(env, posix_errno);
+    }
+
+    // Check if the third argument is an atom
+    if (enif_get_atom(env, argv[2], atom_name, sizeof(atom_name), ERL_NIF_LATIN1)) {
+        if (strcmp(atom_name, "infinity") == 0) {
+            if ((posix_errno = efile_open(&file_path_in, EFILE_MODE_READ, efile_resource_type, &d_in))) {
+                return posix_error_to_tuple(env, posix_errno);
+            }
+
+            if ((posix_errno = efile_read_handle_info(d_in, &info))) {
+                return posix_error_to_tuple(env, posix_errno);
+            }
+
+            len = (off64_t)info.size;
+            efile_close(d_in, &ignored);
```

Comment:
I was attempting to obtain the file size when the 'infinity' atom is passed as the third argument. I will close the file later because I check if the user passes an integer as the third argument. For now, I can simply open the file, read the size information, and proceed without closing the file immediately.